src/Parsers/MySQL/ASTCreateQuery.cpp

```diff
@@ -0,0 +1,128 @@
+#include <Parsers/MySQL/ASTCreateQuery.h>
+
+#include <Interpreters/StorageID.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/CommonParsers.h>
+#include <Parsers/ExpressionElementParsers.h>
+#include <Parsers/ExpressionListParsers.h>
+#include <Parsers/MySQL/ASTCreateDefines.h>
+#include <Parsers/MySQL/ASTDeclarePartitionOptions.h>
+#include <Parsers/MySQL/ASTDeclareTableOptions.h>
+
+namespace DB
+{
+
+namespace MySQLParser
+{
+
+ASTPtr ASTCreateQuery::clone() const
+{
+    auto res = std::make_shared<ASTCreateQuery>(*this);
+    res->children.clear();
+
+    if (columns_list)
+    {
+        res->columns_list = columns_list->clone();
+        res->children.emplace_back(res->columns_list);
+    }
+
+    if (table_options)
+    {
+        res->table_options = table_options->clone();
+        res->children.emplace_back(res->table_options);
+    }
+
+    if (partition_options)
+    {
+        res->partition_options = partition_options->clone();
+        res->children.emplace_back(res->partition_options);
+    }
+
+    return res;
+}
+
+bool ParserCreateQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ASTPtr table;
+    ASTPtr like_table;
+    ASTPtr columns_list;
+    ASTPtr table_options;
+    ASTPtr partition_options;
+    bool is_temporary = false;
+    bool if_not_exists = false;
+
+    if (!ParserKeyword("CREATE").ignore(pos, expected))
+        return false;
+
+    if (ParserKeyword("TEMPORARY").ignore(pos, expected))
+        is_temporary = true;
+
+    if (!ParserKeyword("TABLE").ignore(pos, expected))
+        return false;
+
+    if (ParserKeyword("IF NOT EXISTS").ignore(pos, expected))
+        if_not_exists = true;
+
+    if (!ParserCompoundIdentifier(true).parse(pos, table, expected))
+        return false;
+
+    if (ParserKeyword("LIKE").ignore(pos, expected))
+    {
+        if (!ParserCompoundIdentifier(true).parse(pos, like_table, expected))
+            return false;
+    }
+
+    if (ParserToken(TokenType::OpeningRoundBracket).ignore(pos, expected))
```

Comment:
It will successfully parse queries like `CREAT TABLE t1 LIKE t2 (col1 TYPE1, ...)` or `CREAT TABLE t1 LIKE t2 (LIKE t3)`. Is it expected?